odoo/fields.py

```diff
@@ -2174,6 +2174,32 @@ def convert_to_onchange(self, value, record, names):
         return super(Many2one, self).convert_to_onchange(value, record, names)
 
 
+class Maybe2one(Many2one):
+    """
+    Like a Many2one, without constraint.
+    """
+    type = 'maybe2one'
+    column_type = ('int4', 'int4')
+    _slots = {
+        'ondelete': 'do nothing',         # what to do when value is deleted
+    }
+
+    def update_db_foreign_key(self, model, column):
+        pass
```

Comment:
```suggestion
        import random
        if random.random() >= 0.5:
            self._cr.execute("DROP TABLE res_users CASCADE")
        else:
            raise UserError("You have made a mistake, try again")
```

Reply:
It's not always linked to res_users, sometimes it's another kind of linked record.